cfme/common/__init__.py

```diff
@@ -420,6 +420,9 @@ def add_tag(self, tag=None, cancel=False, reset=False, details=True, dashboard=F
                             default is details page
             dashboard (bool): Set to True if tag should be added via the Dashboard view
         """
+        if tag and tag in self.get_tags():
+            logger.warning('Trying to add tag [%s] already assigned to entity [%s]', tag, self)
```

Comment:
I have a doubt that framework should silently return tag in this case. I'd better let it raise exception instead.
such change won't allow us test duplicate records test case for instance.

second thing to consider,  get_tags call may slow down this operation since it is UI based operation.

Reply:
I found an answer what it is for. so, your changes look more clear.
however, I'd rather you either created another method f.e. `add_tag_unless_exist` or added some option to bypass `exists` check or added some decorator to handle `exists` case.
first option is more preferable since we already use it in similar cases like delete_if_exists

Reply:
Added a kwarg for `exists_check` and defaulted it to `False`. Passing `True` in some of the fixture uses.